### PR TITLE
Add configuration for taking ```max``` of three arguments.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Separate PR for ```max(x,y,z)``` referred in #669

### CODE REVIEW

File: src/runtime/pure/lfortran_intrinsic_math2.f90 [hunk 1]
```diff
@@ -423,6 +423,12 @@ elemental real(dp) function dmax(x, y) result(r)
 end if
 end function
 
+elemental real(dp) function dmax_3args(x,y,z) result(r)
+real(dp), intent(in) :: x,y,z
+r=dmax(x,y)
+r=dmax(r,z)
+end function
```
Comment (czgdp1807) [inline src/runtime/pure/lfortran_intrinsic_math2.f90:430]:
```suggestion
elemental real(dp) function dmax_3args(x,y,z) result(r)
    real(dp), intent(in) :: x, y, z
    r = dmax(x, y)
    r = dmax(r, z)
end function
```

File: tests/max1.f90 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+program max1
+double precision zero, one, two, s
+s=max(abs(zero),abs(one),abs(two))
```
Comment (czgdp1807) [inline tests/max1.f90:3]: May be some more strong tests? For negative numbers as inputs, negatives and positives mixed, etc. Just one test may not be a good idea. Plus please check if they are working by doing an if check something like below,

```suggestion
s=max(abs(zero),abs(one),abs(two))
if(abs(s - abs(two)) > 1e-12) error stop 
```

Comment (Pranavchiku) [reply tests/max1.f90:3]: Yeah sure! I will be doing this in a moment :)

Comment (Pranavchiku) [reply tests/max1.f90:3]: Done with this :)

File: tests/reference/asr-max1-99bbd87.stderr [hunk 3]
```diff
@@ -0,0 +1,5 @@
+semantic error: Arguments do not match for any generic procedure, max
+ --> tests/max1.f90:3:3
+  |
+3 | s=max(abs(zero),abs(one),abs(two))
+  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (certik) [inline tests/reference/asr-max1-99bbd87.stderr:5]: I think this error should not be here?

File: tests/reference/asr-max2-1a7abed.stderr [hunk 4]
```diff
@@ -0,0 +1,5 @@
+semantic error: Arguments do not match for any generic procedure, max
+ --> tests/max2.f90:3:5
+  |
+3 | s = max(x, y, z)
+  |     ^^^^^^^^^^^^
```
Comment (certik) [inline tests/reference/asr-max2-1a7abed.stderr:5]: This error should not be here?

File: tests/reference/asr-max3-f853049.stderr [hunk 5]
```diff
@@ -0,0 +1,6 @@
+syntax error: Token '!=32) error stop
+' (of type 'comment') is unexpected here
+ --> tests/max3.f90:7:5
+  |
+7 | if(s!=32) error stop
+  |     ^^^^^^^^^^^^^^^^^
```
Comment (certik) [inline tests/reference/asr-max3-f853049.stderr:6]: Should not be here?

**Comment by certik** [comment]
Do this:
```
rm -rf tests/reference/*
./run_tests -u
git add tests/reference/*
```
This should make sure all the reference tests are current.

File: tests/reference/asr-max1-99bbd87.stderr (hunk 3 above)
Comment (Pranavchiku) [reply tests/reference/asr-max1-99bbd87.stderr:5]: Yeah I was thinking about it, I will again check it

Comment (certik) [reply tests/reference/asr-max1-99bbd87.stderr:5]: Do the `rm -rf ... run_tests .. git add` trick again. As you can see, the `asr-max1-99bbd87.json` file does not reference this output, so I don't think it's needed.

Comment (Pranavchiku) [reply tests/reference/asr-max1-99bbd87.stderr:5]: Yeah sure will get it done tomorrow morning(my time)

**Comment by Smit-create** [comment]
These changes were already in #675

File: src/runtime/pure/lfortran_intrinsic_math2.f90 [hunk 6]
```diff
@@ -41,7 +41,7 @@ module lfortran_intrinsic_math2
 end interface
 
 interface max
-    module procedure imax, imax8, imax16, imax64, smax, dmax, imax_6args
+    module procedure imax, imax8, imax16, imax64, smax, dmax, imax_6args, dmax_3args
```
Comment (czgdp1807) [inline src/runtime/pure/lfortran_intrinsic_math2.f90:44]: This is already present in `main` branch https://github.com/lfortran/lfortran/blob/main/src/runtime/pure/lfortran_intrinsic_math2.f90?

I think this branch isn't updated yet with `main`. @Smit-create Can you confirm?

Comment (Smit-create) [reply src/runtime/pure/lfortran_intrinsic_math2.f90:44]: yep, that was already done in #675

**Comment by Pranavchiku** [comment]
Closed in reference to #675
